Created 502250 FFC series

jhol opened the pull request.

jhol committed:
Created 502250 FFC series

poeschlr commented:
Did you give the pads for mounting holes a pad number?
Please don't do that. Have a look at other SMD connectors of this lib for guidance.
Example: Molex_Panelmate-53780-0270_02x1.25mm_Angled

---

Are these footprints script generated? (Mentioning this would make checking easier. If they are not i need to check every footprint in freecad.)

---

Naming convention:
The new naming convention for manufacturer specific connectors is as follows:
I marked optional parts with {}
[MAN]_[Series]_[MPN]_[Pin configuration]_P[Pitch]{_Pad[Pad size]}_[Orientation]{_[Options]}


- [Pin layout] - dependent on the connector type
  -  Single row connectors: 1x[number of pins]
  -  Multi row connectors with equal number of pins in each row: [number of rows]x[number of pins per row]
  -  Other connectors: [number of rows]Rows_[number of pins]
  -  Connectors for standardized interfaces do not need to include the pin configuration information (e.g. USB)
- Number format:
  -  [number of rows] number of rows no leading zero.
  -  [number of pins per row] and [number of pins] two digit number with leading zero. (can have more digits for connectors with > 100 pins)

[Orientation] - The orientation of the connector with reference to the board plane (Horizontal or Vertical)
![connector_orientation](https://user-images.githubusercontent.com/18327350/29740362-11afcd9e-8a55-11e7-8856-42fa4ea85819.png)

or because these connectors don't include a mating connector, the better picture is the one normally used for terminal blocks: (The direction of the cable determines the naming)
<img src="https://user-images.githubusercontent.com/18327350/29740365-2da2d3f2-8a55-11e7-9b69-a71710a00e66.png" width=50%/>

This would mean that `Molex_FFC-502244-1530` should be named `Molex_FFC-502244-1530_1x15_P0.5mm_Horizontal`

poeschlr commented:
Is there a reason for the keepout area? It is not mentioned in the datasheet. (Only the single row connector has a keepout area mentioned.)

jhol commented:
@poeschlr

* I previously removed the mounting pad names from the Molex connectors, and removed the pins from the symbols I originally submitted.
* Also, previously I added the single-row symbols to make a complete FFC selection in the symbol library, because I thought that the anchhor/mounting/flange pins were necessary. Given that they are not necessary, the existing generic single-row Conn symbols are sufficient, so I removed all the anchor pins, and all the single row symbols from my merge request.
* The footprints are script generated, so everything will be consistent.
* I'll update the naming to match the scheme you specify - I think it should be `Molex_FFC-502250-1791_2Rows_15_P0.3mm_Horizontal`, right? The pitch is 0.3mm, between odd and even pins.
* I put the keepout area in the area underneath where the latch goes when in the closed position. You wouldn't want to put components in this area. Is this correct usage of the keepout area?

jhol commented:
@poeschlr I've updated the names to reflect the scheme you described.

poeschlr commented:
The keepout area is to denote space where no copper should exist. (It is our current way to tell the user that they should place a keepout zone there.)

The area where no components should be placed is the area within the courtyard outline. So it might be a good idea to just enlarge the courtyard rectangle on this side of the connector.

jhol commented:
@poeschlr Ok, I'll remove the keep out.

jhol commented:
@poeschlr Just pushed a version with the KEEPOUT area removed.

evanshultz commented:
A few notes:
- I don't think `FFC` needs to be in the footprint names as `502250` uniquely describes these parts.
- I don't think we have a paradigm for dual row connectors with an odd number of pins, but `2Rows_17` seems awfully wordy for the 17 pin version. `1x17` nor `2xXX`, the KLC v3 standard, works well. How about `8-7`? Still seems fairly bad but I can't think of anything better right now.
- Are these headers script-generated?

jhol commented:
@evanshultz thanks for your comments...

1. I can remove the FFC text. I just had it there because other groups have a name for the product series, and also there are other FFC footprints in the library. Shall I go ahead and remove it?
2. Can you argue it out with @poeschlr ? I'm happy to do whatever. Just want to get the bike shedding done as soon as possible. 
3. And yes these were generated with a script, so it's easy for me to make changes.

poeschlr commented:
@jhol, @evanshultz
From KLC v3 (see: https://github.com/KiCad/kicad-library/issues/1447)
The series name is required (groups all footprints of the same series together and makes searching easier.)
Also _2Rows_[number of pins ] is our current suggestion in the new KLC. (nobody came forward with a better suggestion)
@SchrodingersGat should we add "Pins" after [number of pins ] (might be a good idea in all cases not just for these type of connector. So also 1x05Pins, _2x05Pins)

evanshultz commented:
Where is `FFC` in the series? I don't see it. FFC is the type of connector, but not the series name.

[Website](http://www.molex.com/molex/products/datasheet.jsp?part=active/5022502791_FFC_FPC_CONNECTORS.xml&channel=Products&Lang=en-US):
![image](https://user-images.githubusercontent.com/1936989/32283162-d688b800-bee0-11e7-8d89-e6e629d95b08.png)

[Datasheet](http://www.molex.com/pdm_docs/sd/5022502791_sd.pdf):
![image](https://user-images.githubusercontent.com/1936989/32283226-05ced82e-bee1-11e7-820e-e2552f4cc995.png)

I don't see FFC listed, but `502250` is shown everywhere as the series name. And it uniquely identifies this family of parts whereas `FFC` applies to a whole range of parts but doesn't add clarity here.

poeschlr commented:
I would say FFC is the human readable version
Including FFC keeps all parts together that are similar. (Even if their series numbers are quite different.)

We include PicoBlade, Micro-Fit, ... even thought their true series "name" is a number.
http://www.molex.com/molex/products/datasheet.jsp?part=active/0530470210_PCB_HEADERS.xml
There are also at least 4 series numbers for picoblade (THT/SMD and for each vertical/horizontal)
This means it is not guaranteed that all picoblade connectors are grouped in the footprint browser if we remove PicoBlade from the name.

P.S.: Why do i use picoblade as an example here? Well these are my de facto standard all purpose connectors. (I use them in all my boards. I still don't know the series number for them. I know they are called picoblade.)

---

Do long file names really bother you that much? (You seem to be the polar opposite to @jkriege2  in this regard.)

evanshultz commented:
@poeschlr I assume you'd rather move discussion about footprint naming over to https://github.com/KiCad/kicad-library/issues/1447#event-1215754526, so I've done that. Merge whenever you're ready.

poeschlr commented:
The KLC defines that the series name should be included. The only question is: What do we mean with series name when we look at Molex connectors.

For me the stuff listed under series is just part of the MPN. (The first "half" of it) The series name is the human readable thing listed under overview/category.

poeschlr commented:
@jhol you say these are script generated. Are your scripts somewhere online? Did you use the [kicad-footprint-generator](https://github.com/pointhi/kicad-footprint-generator) framework by @pointhi?

SchrodingersGat commented:
@jhol we have marked this repository as read-only as we transfer footprints to the new footprints library.

Please resubmit this PR to https://github.com/kicad/kicad-footprints

Thanks, and apologies for the inconvenience
